src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -372,13 +372,21 @@ BookStep<TIn, TOut>::revImp (
         }
     }
 
-
-    if (remainingOut < beast::zero)
+    switch(remainingOut.signum())
     {
-        // something went very wrong
-        assert (0);
-        cache_.emplace (beast::zero, beast::zero);
-        return {beast::zero, beast::zero};
+        case -1:
+        {
+            // something went very wrong
+            assert (0);
```

Comment:
If this is an exceptional case - something that indicates a logic error in the code - then I'd like to see some logging here instead of a simple `assert`.